Gemfile

```diff
@@ -15,6 +15,7 @@ gem "rails-i18n"
 gem "config"
 gem "bootstrap-sass", "3.3.7"
 gem "jquery-rails"
+gem "bcrypt",         "3.1.12"
```

Comment:
gem này dùng làm j

Reply:
dùng để tạo chuỗi mật khẩu được mã hóa an toàn so sánh với mật khẩu gốc với cùng một phương pháp thông qua hằng số cost trong hàm 
    def digest string
      cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
      BCrypt::Engine.cost
      BCrypt::Password.create string, cost: cost
    end